odoo/service/server.py

```diff
@@ -434,6 +438,45 @@ def target():
             t.start()
             _logger.debug("cron%d started!" % i)
 
+    def async_thread(self, n):
+        from odoo.addons.base.models.ir_async import ir_async
+        conn = odoo.sql_db.db_connect('postgres')
+
+        with conn.cursor() as cr:
+            pg_conn = cr._cnx
+            cr.execute("LISTEN odoo_async")
+            cr.commit()
```

Comment:
Does the cursor need to remain open for the rest to work properly?

Reply:
Not sure about the cursor itself but we definitely need to keep the underlying connection (`pg_conn`) to the `postgres` database open

Reply:
Ah yes I see. I thought I'd seen an other place where the cursor is closed but the other one just uses raw pg connections always.